docs/Configuration.md

```diff
@@ -0,0 +1,288 @@
+
+## Table of contents
+- [Project Configuration](#project-configuration)
+- [Extension Configuration](#extension-configuration)
+	+ [Tasks](#tasks)
+	+ [Project Shims](#project-shims)
+- [Specification Versions](#specification-versions)
+
+## Project Configuration
+Typically located in a `ui5.yaml` file per project.
+
+The preferred way of storing the configuration of a UI5 Build and Development Tooling based project is a [YAML](https://yaml.org/) file named `ui5.yaml` inside that project.
+
+**Example**
+````yaml
+---
+specVersion: "0.1"
+type: application
+metadata:
+  name: my-application
+````
+
+Configuration can also be supplied by other means. In cases where a JSON is used, you can still rely on the structure described below.
+
+### Structure
+
+````yaml
+---
+specVersion: "0.1"
+type: application|library|custom
+metadata:
+  name: testsuite
+  copyright: |-
+   <project name>
+    * (c) Copyright 2009-${currentYear} <my company> <my license>
+resources:
+  configuration:
+    paths:
+        "<virtual path>": "<physical path>"
+        "<virtual path 2>": "<physical path 2>"
+builder:
+  customTasks:
+    - name: custom-task-name-1
+      beforeTask: standard-task-name
+      configuration:
+        configuration-key: value
+    - name: custom-task-name-2
+      afterTask: custom-task-name-1
+      configuration:
+        color: blue
+server:
+  settings:
+    port: 8099
+````
+
+### Properties
+#### \<root\>
+- `specVersion`: Version of the specification this configuration is based on. See [Specification Versions](#specification-versions).
+- `type`: Either `application`, `library` or `custom` (custom not yet implemented); defines the default path mappings and build steps. Custom doesn't define any specific defaults.
+
+#### metadata
+Some general information:
+- `name`: Name of the application/library/resource
+- `copyright` (optional): String to be used for replacement of copyright placeholders in the project
+
+#### resources (optional)
+- `configuration`
+    - `paths`: Mapping between virtual and physical paths
+        + For type `application` there can be a setting for mapping the virtual path `webapp` to a physical path within the project
+        + For type `library` there can be a setting for mapping the virtual paths `src` and `test` to physical paths within the project
+
+#### builder (optional)
+- `customTasks` (optional, list): In this block, you define additional custom build tasks, see [here](./BuildExtensibility.md) for a detailed explanation and examples of the build extensibility. Each entry in the `customTasks` list consists of the following options:
+  - `name` (mandatory): The name of the custom task
+  - `afterTask` or `beforeTask` (only one, mandatory): The name of the build task after or before which your custom task will be executed.
+  - `configuration` (optional): Additional configuration that is passed to the custom build task
+
+#### server (optional)
+- `settings` (not yet implemented)
```

Comment:
Indeed not implemented, yet. Do we have an open issue for that?

Reply:
Yeah, this just needs to be done. I'll create an issue for it.

![JUST DO IT](https://media.giphy.com/media/GcSqyYa2aF8dy/giphy.gif)

Reply:
Created https://github.com/SAP/ui5-project/issues/89